etc/gcc-flags.cmake

```diff
@@ -3,11 +3,29 @@ include_guard(GLOBAL)
 set(CMAKE_CXX_STANDARD 23)
 
 set(CMAKE_CXX_FLAGS
-   "-std=c++23 -Wall -Wextra "
-CACHE STRING "CXX_FLAGS" FORCE)
-
-set(CMAKE_CXX_FLAGS_DEBUG "-O0 -fno-inline -g3" CACHE STRING "C++ DEBUG Flags" FORCE)
-set(CMAKE_CXX_FLAGS_RELEASE "-Ofast -g0 -DNDEBUG" CACHE STRING "C++ Release Flags" FORCE)
-set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG" CACHE STRING "C++ RelWithDebInfo Flags" FORCE)
-set(CMAKE_CXX_FLAGS_TSAN "-O3 -g -DNDEBUG -fsanitize=thread" CACHE STRING "C++ TSAN Flags" FORCE)
-set(CMAKE_CXX_FLAGS_ASAN "-O3 -g -DNDEBUG -fsanitize=undefined" CACHE STRING "C++ ASAN Flags" FORCE)
+    "-std=c++23 -Wall -Wextra "
+    CACHE STRING "CXX_FLAGS" FORCE)
+
+set(CMAKE_CXX_FLAGS_DEBUG
+    "-O0 -fno-inline -g3"
+    CACHE STRING "C++ DEBUG Flags" FORCE)
+set(CMAKE_CXX_FLAGS_RELEASE
+    "-Ofast -g0 -DNDEBUG"
+    CACHE STRING "C++ Release Flags" FORCE)
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO
+    "-O3 -g -DNDEBUG"
+    CACHE STRING "C++ RelWithDebInfo Flags" FORCE)
+set(CMAKE_CXX_FLAGS_TSAN
+    "-O3 -g -DNDEBUG -fsanitize=thread"
+    CACHE STRING "C++ TSAN Flags" FORCE)
+set(CMAKE_CXX_FLAGS_ASAN
+    "-O3 -g -DNDEBUG -fsanitize=undefined"
+    CACHE STRING "C++ ASAN Flags" FORCE)
+
```

Comment:
I think we should consider having CXX_FLAGS_HARDENED -- have a look at the recommendations here: https://best.openssf.org/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.html